src/core/qgstemporalnavigationobject.cpp

```diff
@@ -346,7 +346,12 @@ long long QgsTemporalNavigationObject::totalFrameCount() const
   else
   {
     const QgsInterval totalAnimationLength = mTemporalExtents.end() - mTemporalExtents.begin();
-    return static_cast< long long >( std::ceil( totalAnimationLength.seconds() / mFrameDuration.seconds() ) );
+    if ( mTemporalExtents.includeEnd() )
+      // adding a fraction here to the totalAnimationLength, instead of always adding a frame,
+      // because having a feature NOT ON the totalAnimationLength.end limit will also get an extra frame
+      return static_cast< long long >( std::ceil( ( totalAnimationLength.seconds() + 0.001 ) / mFrameDuration.seconds() ) );
```

Comment:
@nyalldawson can we disscuss why this is silly? Or if there is another way to NOT add always add an extra frame?
Because I think that the only way to decide here IF we need an extra frame would be to use the last filter frame on alle temporal layers and THEN decide.
To show the case: THIS example should return 4 frames (aka 1 extra), as the last measurement is ON the end of the animationLength:

```
       # map range              1----2----3----4--------hours
       # features               .    .    .
```
While this one should return 3 frames, because there is no data ON the end:
```
       # map range              1----2----3----4--------hours
       # features               .    . .   
```
Anybody has an idea to handle this, other then always return 
    `1 + ( std::ceil( totalAnimationLength.seconds() / mFrameDuration.seconds() ) )`
?